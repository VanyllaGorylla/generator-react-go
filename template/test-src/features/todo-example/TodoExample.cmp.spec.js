import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import moxios from 'moxios';

import WrappedTodoExample, {
  TodoExample
} from '@/features/todo-example/TodoExample.cmp';

describe('TodoExample test', function() {
  let store;

  beforeEach(() => {
    moxios.install();
    store = configureStore()({
      todos: [
        {
          id: 1,
          title: 'Do Laundry',
          completed: false
        },
        {
          id: 2,
          title: 'Wash your dishes',
          completed: true
        },
        {
          id: 3,
          title: 'Wash the floor',
          completed: false
        }
      ]
    });
  });

  it('should render proper html', function() {
    let addTodo = jest.fn();

    const todos = [
      {
        id: 1,
        title: 'Do Laundry',
        completed: false
      },
      {
        id: 2,
        title: 'Wash your dishes',
        completed: true
      },
      {
        id: 3,
        title: 'Wash the floor',
        completed: false
      }
    ];

    const tree = renderer
      .create(
        <Provider store={store}>
          <WrappedTodoExample />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(function() {
    moxios.uninstall();
  });
});
