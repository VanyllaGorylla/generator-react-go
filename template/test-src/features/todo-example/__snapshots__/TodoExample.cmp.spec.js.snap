// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoExample test should render proper html 1`] = `
<div
  className="container"
>
  <div
    className="card"
  >
    <div
      className="card-content"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        Todo Example
      </span>
      <div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <form
              onSubmit={[Function]}
            >
              <input
                onChange={[Function]}
                placeholder="Put todo name here"
                type="text"
                value=""
              />
              <button
                className="btn"
                type="submit"
              >
                Add todo
              </button>
            </form>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col"
          >
            <ul>
              <li
                onClick={[Function]}
              >
                <input
                  defaultChecked={false}
                  id="li-checkbox-1"
                  onChange={[Function]}
                  onClick={[Function]}
                  type="checkbox"
                />
                <label
                  className="label-color"
                  htmlFor="li-checkbox-1"
                  style={
                    Object {
                      "textDecoration": "",
                    }
                  }
                >
                  Do Laundry
                </label>
              </li>
              <li
                onClick={[Function]}
              >
                <input
                  defaultChecked={true}
                  id="li-checkbox-2"
                  onChange={[Function]}
                  onClick={[Function]}
                  type="checkbox"
                />
                <label
                  className=""
                  htmlFor="li-checkbox-2"
                  style={
                    Object {
                      "textDecoration": "line-through",
                    }
                  }
                >
                  Wash your dishes
                </label>
              </li>
              <li
                onClick={[Function]}
              >
                <input
                  defaultChecked={false}
                  id="li-checkbox-3"
                  onChange={[Function]}
                  onClick={[Function]}
                  type="checkbox"
                />
                <label
                  className="label-color"
                  htmlFor="li-checkbox-3"
                  style={
                    Object {
                      "textDecoration": "",
                    }
                  }
                >
                  Wash the floor
                </label>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
    <div
      className="card-reveal"
    >
      <span
        className="card-title grey-text text-darken-4"
      >
        Todo Example
        <i
          className="material-icons right"
        >
          close
        </i>
      </span>
    </div>
  </div>
</div>
`;
